pennylane/circuit_graph.py

```diff
@@ -160,6 +160,14 @@ def __init__(self, ops, variable_deps, wires):
         """int: number of wires the circuit contains"""
         for k, op in enumerate(ops):
             op.queue_idx = k  # store the queue index in the Operator
+
+            if hasattr(op, "return_type") and op.return_type is qml.operation.State:
```

Comment:
@josh146 went eventually with making up for the changes needed to the wires of state here. Lmk if you think this is fine or it's better to modify.

Reply:
@antalszava I think this is a much improved solution :)